kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/InternalCompletableFuture.java

```diff
@@ -10,21 +10,27 @@
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import io.netty.channel.EventLoop;
 
 /**
- * Implementation of CompletableFuture that uses the given {@link Executor}
- * for async method invocations that do not specify one.
+ * Implementation of CompletableFuture that executes all chained work on a specific
+ * event loop.
  * <br/>
  * @param <T> The result type returned by this future's {@code join}
  * and {@code get} methods.
  * @see InternalCompletionStage
  */
 class InternalCompletableFuture<T> extends CompletableFuture<T> {
 
-    private final Executor executor;
+    private final EventLoop eventLoop;
```

Comment:
note to self:  I've not understood the reason for the switch from Executor to Netty Event Loop yet.

Reply:
ah, it is so we can tell if we are on the event loop or not

Reply:
Would ThreadAwareExecutor be a  option?